eventbridge.tf

```diff
@@ -1,20 +1,43 @@
-resource "aws_cloudwatch_event_rule" "wildcard" {
-  name_prefix = local.name_prefix
-  description = "Capture all events in account"
-  event_pattern = jsonencode({
-    "account" : [data.aws_caller_identity.current.account_id]
-  })
-  tags = var.tags
+locals {
+  eventbridge_rules = var.eventbridge_rules != null ? var.eventbridge_rules : [
+    {
+      description = "Capture all events other than Cloudtrail"
+      event_pattern = jsonencode({
+        detail-type = [{ anything-but = "AWS API Call via CloudTrail" }]
+      })
+    },
+    {
+      description = "Capture all Cloudtrail events"
+      event_pattern = jsonencode({
+        detail-type = ["AWS API Call via CloudTrail"]
+        detail = {
+          eventSource = [{ anything-but = var.cloudtrail_exclude_management_event_sources }]
+        }
+      })
+    }
+  ]
+}
+
+resource "aws_cloudwatch_event_rule" "rules" {
+  count         = length(local.eventbridge_rules)
```

Comment:
Use `for_each` here to avoid replacements if a rule is inserted between existing rules. Options for this would be:

* Make the input a map instead, and document that the keys are only used in resource addresses
* Keep the list but put `id` or similar on the object and then internally turn it it into a map in the `for_each` expression, using a for comprehension

Reply:
I assume if I were to use `id`, I'd need to use a `validation` rule to verify there are no dupes?

Reply:
Yes, which might get kinda cumbersome. Makes some sense for something like `Sid` in IAM where you actually have an ID key that requires uniqueness.

Reply:
I could use the `description`? It becomes slightly awkward in that changing the description would recreate the rule, but perhaps that's not too terrible.

I'm more hesitant to use a map, because the module further ahead takes a list of rules, and so ordering is important.

Reply:
Yes, but at that point I wouldn't bother, as the idea that the description is the primary is more unexpected than the list order.

The called module seems to have the same issue, the order is not fundamentally important, it's just also using `count`. 

Once you accept an input from users, I'd say this hazard is worth closing.

Reply:
We might as well just fix this in the called module: https://github.com/observeinc/terraform-aws-kinesis-firehose/pull/30

Reply:
Ok, updated commit with new underlying module. Much cleaner.